src/commands/serve/serve.mjs

```diff
@@ -0,0 +1,184 @@
+    log(`${NETLIFYDEVLOG} Injecting environment variable values for ${chalk.yellow('all scopes')}`)
+  }
+
+  await injectEnvVariables({ devConfig, env, site })
+  await promptEditorHelper({ chalk, config, log, NETLIFYDEVLOG, repositoryRoot, state })
+
+  const { addonsUrls, capabilities, siteUrl, timeouts } = await getSiteInformation({
+    // inherited from base command --offline
+    offline: options.offline,
+    api,
+    site,
+    siteInfo,
+  })
+
+  // Ensure the internal functions directory exists so that the functions
+  // server and registry are initialized, and any functions created by
+  // Netlify Build are loaded.
+  await getInternalFunctionsDir({ base: site.root, ensureExists: true })
+
+  /** @type {Partial<import('../../utils/types').ServerSettings>} */
+  let settings = {}
+  try {
+    settings = await detectServerSettings(devConfig, options, site.root)
+
+    cachedConfig.config = getConfigWithPlugins(cachedConfig.config, settings)
+  } catch (error_) {
+    log(NETLIFYDEVERR, error_.message)
+    exit(1)
+  }
+
+  command.setAnalyticsPayload({ projectType: settings.framework || 'custom', live: options.live, graph: options.graph })
+
+  log(`${NETLIFYDEVWARN} Building site for production`)
+
+  const { configPath: configPathOverride } = await runBuildTimeline({ cachedConfig, options, settings, site })
+
+  await startFunctionsServer({
+    api,
+    command,
+    config,
+    debug: options.debug,
+    loadDistFunctions: true,
+    settings,
+    site,
+    siteInfo,
+    siteUrl,
+    capabilities,
+    timeouts,
+  })
+
+  // Try to add `.netlify` to `.gitignore`.
+  try {
+    await ensureNetlifyIgnore(repositoryRoot)
+  } catch {
+    // no-op
+  }
+
+  // TODO: We should consolidate this with the existing config watcher.
+  const getUpdatedConfig = async () => {
+    const cwd = options.cwd || process.cwd()
+    const { config: newConfig } = await command.getConfig({ cwd, offline: true, state })
+    const normalizedNewConfig = normalizeConfig(newConfig)
+
+    return normalizedNewConfig
+  }
+
+  const inspectSettings = generateInspectSettings(options.edgeInspect, options.edgeInspectBrk)
+  const url = await startProxyServer({
+    addonsUrls,
+    config,
+    configPath: configPathOverride,
+    env,
+    geolocationMode: options.geo,
+    geoCountry: options.country,
+    getUpdatedConfig,
+    inspectSettings,
+    offline: options.offline,
+    settings,
+    site,
+    siteInfo,
+    state,
+  })
+
+  if (devConfig.autoLaunch !== false) {
+    await openBrowser({ url, silentBrowserNoneError: true })
+  }
+
+  process.env.URL = url
+  process.env.DEPLOY_URL = url
+
+  printBanner({ url })
+}
+
+/**
+ * Creates the `netlify serve` command
+ * @param {import('../base-command.mjs').default} program
+ * @returns
+ */
+export const createServeCommand = (program) =>
+  program
+    .command('serve')
+    .description('(Beta) build the site for production and serve locally')
```

Comment:
```suggestion
    .description('(Beta) Build the site for production and serve locally. This does not watch the code for changes, so if you need to rebuild your site then you must exit and run `serve` again.')
```
I dunno if we need it now, but it would be good to have a sentence or so explaining what's going on here. Are the docs generated from this?

Reply:
Yeah, https://cli.netlify.com is. I've added your commit and updated docs in https://github.com/netlify/cli/pull/5351/commits/ad1dbf4b6d3ae147b1db9f0afae664a21207d85c. Also added a log message in https://github.com/netlify/cli/pull/5351/commits/431e531cc54270650741affacd814dacf5b65d11.